### PR TITLE
Rationalize the ScrollMetrics cloning logic

### PR DESCRIPTION
**Pull request by Hixie**

### CODE REVIEW

**Comment by Hixie** [comment]
cc @xster

**Comment by Hixie** [comment]
friendly ping :-)

File: packages/flutter/lib/src/widgets/scroll_metrics.dart [hunk 1]
```diff
@@ -105,11 +121,22 @@ class FixedScrollMetrics extends ScrollMetrics {
...
-    pixels = parent.pixels,
-    viewportDimension = parent.viewportDimension,
+  ///
+  /// The named arguments allow the values to be optionally adjusted in the
+  /// process. See also [cloneMetrics].
+  ///
+  /// Values that are null in the given `parent` will have any specified deltas
+  /// ignored (the value remains null in the clone).
+  FixedScrollMetrics.clone(ScrollMetrics parent, {
+    double deltaMinScrollExtent: 0.0,
```
Comment (xster) [inline packages/flutter/lib/src/widgets/scroll_metrics.dart]: In retrospect, I guess letting the null-ness of the fields be directly inferred from the null-ness of the constructor arguments is clearer than the null-ness of one of the fields of the parent (base on chat about copyWith)

**Review by xster — APPROVED** [review]
LG

File: packages/flutter/lib/src/widgets/scroll_metrics.dart (hunk 1 above)
Comment (Hixie) [reply packages/flutter/lib/src/widgets/scroll_metrics.dart]: I'll submit a patch that does the copyWith thing instead.

**Comment by xster** [comment]
LG

**Pull request merged**